Library/Homebrew/cask/cask.rb

```diff
@@ -96,31 +107,28 @@ def refresh
       define_method(method_name) { |&block| @dsl.send(method_name, &block) }
     end
 
-    sig { returns(T::Array[[String, String]]) }
-    def timestamped_versions
-      relative_paths = Pathname.glob(metadata_timestamped_path(version: "*", timestamp: "*"))
+    sig { params(caskroom_path: Pathname).returns(T::Array[[String, String]]) }
+    def timestamped_versions(caskroom_path: self.caskroom_path)
+      relative_paths = Pathname.glob(metadata_timestamped_path(
+                                       version: "*", timestamp: "*",
+                                       caskroom_path: caskroom_path
+                                     ))
                                .map { |p| p.relative_path_from(p.parent.parent) }
       # Sorbet is unaware that Pathname is sortable: https://github.com/sorbet/sorbet/issues/6844
       T.unsafe(relative_paths).sort_by(&:basename) # sort by timestamp
        .map { |p| p.split.map(&:to_s) }
     end
 
-    def versions
-      timestamped_versions.map(&:first)
-                          .reverse
-                          .uniq
-                          .reverse
-    end
```

Comment:
This is completely beside the point but I find it interesting there's absolutely no public API for dealing with casks.

Is dealing with casks just not allowed at all in third-party tools?

IMO, `homebrew-bundle` is an example of something that should be able to work with public API only. It is after all an example of something that was third-party made first-party, so seems silly for it to be something that third-party developers shouldn't be allowed to do.

(Sorry for the off-topic, maybe worth creating a separate discussion for this)